### PR TITLE
improves CITF code

### PR DESCRIPTION
**Pull request by a4abhishek**
- introduced Debug in configuration
- decoupled util package from rest of the packages
- Updated `GetContainerStateInPod` to `GetContainerStatesInPod` which will return states of all the containers instead of first one
- New method `GetContainerStateByIndexInPod` which will return state of container which has that index in the list
- Created a new method `GetContainerStatesInPodUntilToldToQuit` which will try to get the states of container until signalled to quit
- Created a new method `GetContainerStatesInPodWithTimeout` which will try to get the states of the container until timeout hit
- Created a new method `GetContainerStateByIndexInPodUntilToldToQuit` which is indexed version of `GetContainerStatesInPodUntilToldToQuit`
- Created a new method `GetContainerStateByIndexInPodWithTimeout` which is indexed version of `GetContainerStatesInPodWithTimeout`
- Updated `GetPods` so that instead of trying to get the pod for some time, it tries only once.
- Created a new method `GetPodsUntilQuitSignal` which will try to get the pods which match the criteria unless it gets at-least one or signalled to quit
- Created a new method `GetPodOrTimeout` which will try to get the pods which match the criteria unless it gets at-least one or timeout hit.
- Created a new method `GetPodsOrBlocked` which tries indefinitely until it gets at least one pod which match the criteria
- Updated `GetPod` method which now takes exact name of pod and return it's client-go structural representation.
- Created a new method `BlockUntilPodIsUp` which blocks the control untill the given pod is ready
- Created a new method `BlockUntilPodIsUpOrTimeout`which blocks the control until the given pod is ready or timeout is hit
etc.

Changes committed:
	modified:   citf.go
	deleted:    common/common.go
	modified:   config/config.go
	modified:   environments/minikube/setup.go
	modified:   utils/k8s/check_status.go
	modified:   utils/k8s/general.go
	modified:   utils/k8s/k8s.go
	modified:   utils/string/string.go
	modified:   utils/system/exec.go
	modified:   utils/system/general.go
	modified:   utils/system/run.go

Signed-off-by: Abhishek Kashyap <abhishek.kashyap@mayadata.io>

### CODE REVIEW

File: config/config.go [hunk 1]
```diff
@@ -36,9 +40,18 @@ var defaultConf Configuration
 func init() {
 	defaultConf = Configuration{
 		Environment: "minikube",
+		Debug:       false,
```
Comment (shovanmaity) [inline config/config.go]: we can have something like that
```
const (
	defaultEnabled  = true 
	defaultDisabled = false
)
```

File: config/config.go [hunk 2]
```diff
@@ -93,3 +109,11 @@ func GetConf(field string) string {
 func Environment() string {
 	return GetConf("Environment")
 }
+
+// Debug returns the environment which should be used in testing
+func Debug() bool {
+	return strings.ToLower(GetConf("Debug")) == "true"
```
Comment (shovanmaity) [inline config/config.go]: same as before.

File: utils/k8s/check_status.go [hunk 3]
```diff
@@ -54,3 +54,14 @@ func (k8s K8S) IsPodStateGood(podState string) bool {
 
 	return false
 }
+
+// IsPodRunning returns if all the containers in a given pod is running
```
Comment (satyamz) [inline utils/k8s/check_status.go]: `IsPodRunning` returns what?

**Comment by satyamz** [comment]
My thought's are inline with @shovanmaity's ! Please consider those suggestions too.

File: citf.go [hunk 4]
```diff
@@ -17,7 +17,6 @@ import (
 	"fmt"
 
 	"github.com/golang/glog"
-	"github.com/openebs/CITF/common"
 	"github.com/openebs/CITF/config"
 	"github.com/openebs/CITF/environments"
```
Comment (princerachit) [inline citf.go]: s

**Comment by a4abhishek** [comment]
@shovanmaity @satyamz addressed your comments, have a look again.

**Comment by a4abhishek** [comment]
I am very thankful to @shovanmaity and @sagarkrsd for enlightening me about benefits of using constants (not variables) instead of bare strings in the code.

**Pull request merged**